modules/light-clients/08-wasm/types/vm.go

```diff
@@ -151,16 +155,19 @@ func wasmSudo[T ContractResult](ctx sdk.Context, cdc codec.BinaryCodec, clientSt
 	}
 
 	checksum := cs.Checksum
-	resp, err := callContract(ctx, clientStore, checksum, encodedData)
+	res, err := callContract(ctx, clientStore, checksum, encodedData)
 	if err != nil {
-		return result, errorsmod.Wrap(ErrWasmContractCallFailed, err.Error())
+		return result, errorsmod.Wrap(ErrVMError, err.Error())
+	}
+	if res.Err != "" {
```

Comment:
I don't feel strongly about either, so I am fine with whatever people prefer. Maybe I am slightly inclined to leave it like this, so that `checkResponse` is limited to checking that the contract doesn't return data that we expect, but I am fine with both.